Source: Rebased from https://bugs.gentoo.org/show_bug.cgi?id=302624
Upstream: no, proprietary upstream.
Reason: Allow rebuilding autotools with a modern version.

diff --git a/backend/Makefile.am b/backend/Makefile.am
index 4a36339..57e0176 100644
--- a/backend/Makefile.am
+++ b/backend/Makefile.am
@@ -58,7 +58,6 @@ noinst_LTLIBRARIES = \
 
 libepkowa_la_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
-	$(LTDLINCL) \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/non-free \
 	$(XML_CFLAGS) \
@@ -70,7 +69,7 @@ libepkowa_la_LIBADD = \
 	-lm \
 	$(XML_LIBS) \
 	$(LIBUSB_1_0_LIBS) \
-	$(LIBLTDL)
+	-lltdl
 libepkowa_la_SOURCES = \
 	$(sane_backends_files) \
 	ipc.c \
diff --git a/configure.ac b/configure.ac
index f28ed38..08354cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -266,13 +266,10 @@ AC_CHECK_FUNCS([\
 dnl  The somewhat obscure LT_AC_PROG_SED is here to make sure that the
 dnl  SED environment variable gets set before it's used by the libtool
 dnl  machinery.  This works around buggy libtool versions.
-LT_AC_PROG_SED
-AC_LIBLTDL_INSTALLABLE
-AC_LIBTOOL_DLOPEN
 AC_PROG_LIBTOOL
-AC_LIB_LTDL
-AC_SUBST(LTDLINCL)
-AC_SUBST(LIBLTDL)
+AC_LTDL_SHLIBPATH
+AC_LTDL_SHLIBEXT
+AC_LTDL_SYSSEARCHPATH
 
 
 dnl  Graphic file format options
diff --git a/frontend/Makefile.am b/frontend/Makefile.am
index 6d7c132..0fa9c81 100644
--- a/frontend/Makefile.am
+++ b/frontend/Makefile.am
@@ -34,7 +34,7 @@ iscan_CXXFLAGS = \
 iscan_LDADD = \
 	$(top_builddir)/lib/libimage-stream.la \
 	-lsane \
-	@LIBLTDL@ \
+	-lltdl \
 	@GTK_LIBS@ \
 	@GDK_IMLIB_LIBS@ \
 	$(top_builddir)/non-free/libesmod.so
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 922ca44..0fa42d4 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -30,7 +30,7 @@ noinst_LTLIBRARIES = libimage-stream.la
 libimage_stream_la_CPPFLAGS = -I$(top_srcdir)/include
 libimage_stream_la_LDFLAGS = -static
 libimage_stream_la_LIBADD  = \
-	$(LIBLTDL) \
+	-lltdl \
 	$(top_builddir)/lib/pdf/libpdf.la
 libimage_stream_la_SOURCES = \
 	$(libimage_stream_la_files)
-- 
1.7.11.4

